### PR TITLE
Debug example code for MegaForCausalLM

### PR DESCRIPTION
**Pull request by Tylersuard**
set ignore_mismatched_sizes=True in model loading code for MegaForCausalLM so that the example code runs without errors.

# What does this PR do?

Fixes # 22974

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.  https://github.com/huggingface/transformers/issues/22974
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@amyeroberts

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing this! 

For the quality checks, you'll need to run `make style` and push the changes. Once all the tests are green we're good to merge :)

**Comment by Tylersuard** [comment]
> Thanks for fixing this!
> 
> For the quality checks, you'll need to run `make style` and push the changes. Once all the tests are green we're good to merge :)

Do I run make style in the main Transformers directory?

**Comment by amyeroberts** [comment]
@Tylersuard Hmmmm..... OK, I'm not exactly sure what happened - I suspect there might be a mismatch in installed package versions. Here's how I would try to resolve:

First undo the most recent changes added to the unrelated files: 
* Undo the last two commits: `git revert --hard HEAD~2`
* Push these changes to the PR: `git push -f`

Then get your branch in sync with `main`: 
* Get the latest version of main: `git checkout main && git fetch upstream main && git rebase upstream/main`
* Install latest formatting settings `pip install -e ".[quality]"`
* Rebase main onto this branch `git checkout patch-1 && git rebase upstream/main`
* Push these changes to the PR (you'll have to force): `git push --force`
* Make any style changes `make style` 
* Commit changes made (should just be to the modeling_mega.py file): `git add src/transformers/models/mega/modeling_mega.py && git commit -m "Fix up" && git push`

**Comment by Tylersuard** [comment]
@amyeroberts Very clear instructions, thank you!

**Comment by Tylersuard** [comment]
@amyeroberts Ok, all done!  I do not have write access, so I can't merge the PR

**Pull request merged**
